src/tests/index_btree_benchmark.cpp

```diff
@@ -0,0 +1,262 @@
+        std::shared_ptr< IndexTableBase > on_index_table_found(const superblk< index_table_sb >& sb) {
+            LOGINFO("Index table recovered");
+            LOGINFO("Root bnode_id {} version {}", sb->root_node, sb->link_version);
+            m_benchmark->m_bt = std::make_shared< typename T::BtreeType >(sb, m_benchmark->m_cfg);
+            return m_benchmark->m_bt;
+        }
+
+    private:
+        IndexBtreeBenchmark* m_benchmark;
+    };
+
+    IndexBtreeBenchmark() {
+        this->m_is_multi_threaded = true;
+        SetUp();
+    }
+
+    ~IndexBtreeBenchmark() { TearDown(); }
+
+    void SetUp() {
+        test_common::HSTestHelper::start_homestore(
+            "index_btree_benchmark",
+            {{HS_SERVICE::META, {.size_pct = 10.0}},
+             {HS_SERVICE::INDEX, {.size_pct = 70.0, .index_svc_cbs = new TestIndexServiceCallbacks(this)}}});
+
+        this->m_cfg = BtreeConfig(hs()->index_service().node_size());
+
+        auto uuid = boost::uuids::random_generator()();
+        auto parent_uuid = boost::uuids::random_generator()();
+
+        BtreeHelper< TestType >::SetUp();
+        this->m_bt = std::make_shared< typename T::BtreeType >(uuid, parent_uuid, 0, this->m_cfg);
+        hs()->index_service().add_index_table(this->m_bt);
+    }
+
+    void TearDown() {
+        BtreeHelper< TestType >::TearDown();
+        test_common::HSTestHelper::shutdown_homestore();
+    }
+
+    void ConcurrentOps(std::vector< std::string > input_ops) {
+        std::vector< std::pair< std::string, int > > ops;
+        if (SISL_OPTIONS.count("operation_list")) {
+            input_ops = SISL_OPTIONS["operation_list"].as< std::vector< std::string > >();
+        }
+        int total = std::accumulate(input_ops.begin(), input_ops.end(), 0, [](int sum, const auto& str) {
+            std::vector< std::string > tokens;
+            boost::split(tokens, str, boost::is_any_of(":"));
+            if (tokens.size() == 2) {
+                try {
+                    return sum + std::stoi(tokens[1]);
+                } catch (const std::exception&) {
+                    // Invalid frequency, ignore this element
+                }
+            }
+            return sum; // Ignore malformed strings
+        });
+
+        std::transform(input_ops.begin(), input_ops.end(), std::back_inserter(ops), [total](const auto& str) {
+            std::vector< std::string > tokens;
+            boost::split(tokens, str, boost::is_any_of(":"));
+            if (tokens.size() == 2) {
+                try {
+                    return std::make_pair(tokens[0], (int)(100.0 * std::stoi(tokens[1]) / total));
+                } catch (const std::exception&) {
+                    // Invalid frequency, ignore this element
+                }
+            }
+            return std::make_pair(std::string(), 0);
+        });
+
+        this->run_in_parallel(ops);
+    }
+
+    void put() { this->ConcurrentOps({"put:100"}); }
```

Comment:
Not sure if we need these functions?

Reply:
the advantage we have a seperate put(), remove() and other functions is if we wan to take some actions more than just passing the operation_list, we can implement this logic inside these functions.

furthermore, if we want to implement a complex benchmark test ,  we just need to the logic in a seperate function and then add the items in the macro.